client-src/elements/chromedash-featurelist.ts

```diff
@@ -8,48 +10,36 @@ const MAX_FEATURES_SHOWN = 500;
 class ChromedashFeaturelist extends LitElement {
   static styles = FEATURELIST_CSS;
 
-  static get properties() {
-    return {
-      user: {type: Object},
-      isSiteEditor: {type: Boolean},
-      signedInUser: {type: String},
-      editableFeatures: {type: Object},
-      features: {attribute: false}, // Directly edited and accessed in template/features.html
-      metadataEl: {attribute: false}, // The metadata component element. Directly edited in template/features.html
-      searchEl: {attribute: false}, // The search input element. Directly edited in template/features.html
-      filtered: {attribute: false},
-      openFeatures: {attribute: false},
-      starredFeatures: {attribute: false},
-    };
-  }
+  @property({type: Object})
+  user!: User;
+  @property({type: Boolean})
+  isSiteEditor = false;
+  @property({type: String})
+  signedInUser = '';
+  @property({type: Array})
+  editableFeatures!: number[];
+  @state()
+  features!: Feature[];
+  /** The search input element. */
+  @state()
+  filtered!: Feature[];
+  @state()
+  openFeatures = new Set<number>();
+  @state()
+  starredFeatures = new Set<number>();
+  searchEl: HTMLInputElement;
+  metadataEl: HTMLElement;
+  _hasInitialized = false; // Used to check initialization code.
+  _hasScrolledByUser = false; // Used to set the app header state.
+  _featuresUnveilMetric = new Metric('features_unveil');
+  _featuresFetchMetric = new Metric('features_loaded');
 
   constructor() {
     super();
-    this.user = {};
-    this.features = [];
-    this.editableFeatures = [];
-    this.filtered = [];
-    this.metadataEl = document.querySelector('chromedash-metadata');
-    this.searchEl = document.querySelector('.search input');
-    this.isSiteEditor = false;
-    this.signedInUser = '';
-    this._hasInitialized = false; // Used to check initialization code.
-    this._hasScrolledByUser = false; // Used to set the app header state.
-    /* When scrollTo(), we also expand the feature. This is the id of the feature. */
-    this.openFeatures = new Set();
-    this.starredFeatures = new Set();
-
-    this._featuresUnveilMetric = new Metric('features_unveil');
-    this._featuresFetchMetric = new Metric('features_loaded');
-    this._featuresUnveilMetric.start();
     this._featuresFetchMetric.start();
-
-    /* The running context `this` inside `renderItem` is `lit-virtualizer`
-     * rather than `chromedash-featurelist`, so all function calls inside need
-     * to be bound to `this`. */
-    this._onFeatureToggledBound = this._onFeatureToggled.bind(this);
-    this._onStarToggledBound = this._onStarToggled.bind(this);
-
+    this._featuresUnveilMetric.start();
+    this.searchEl = document.querySelector('.search input')!;
+    this.metadataEl = document.querySelector('chromedash-metadata')!;
```

Comment:
No need to change this now, but in future changes it can be helpful to keep unchanged lines in the same order as they were in the original code, to minimize the apparent diff for reviewers.